pkg/kratos/service.go

```diff
@@ -123,14 +125,21 @@ func (s *Service) CreateBrowserLoginFlow(
 	ctx, span := s.tracer.Start(ctx, "kratos.Service.CreateBrowserLoginFlow")
 	defer span.End()
 
-	flow, resp, err := s.kratos.FrontendApi().
+	request := s.kratos.FrontendApi().
 		CreateBrowserLoginFlow(ctx).
 		Aal(aal).
 		ReturnTo(returnTo).
-		LoginChallenge(loginChallenge).
 		Refresh(refresh).
-		Cookie(httpHelpers.CookiesToString(cookies)).
-		Execute()
+		Cookie(httpHelpers.
+			CookiesToString(cookies))
+
+	if loginChallenge != "" {
+		request = request.LoginChallenge(loginChallenge)
+	} else if returnTo == "" {
+		return nil, nil, ErrNoReturnToOrLoginChallenge
```

Comment:
```suggestion
		return nil, nil, fmt.Errorf("No return_to or login_challenge was provided")
```

i'd probably drop the const and tweak the tests

Reply:
why drop the const? I find it clearer to use only vars for generating tests, this allows you to gather them to a single place and imo makes it more manageable.

No problem to change it, just curious

Reply:
main reason is we don't currently instantiate error messages this way in the codebase, i m happy to change it everywhere but i would keep it as a separate effort

 in terms of testing, fair point, at the same time, unless we are interested in the error message itself, what we should be checking is that the function errors (or at least that's my opinion)